[DPP-771][Error codes] Add readme.md and supplement docs with a concrete error handling example. (#11990)
digital-asset/daml: docs/source/app-dev/grpc/error-codes.rst

Working with Error Codes
======================================

> The example below shows how a user can in practice extract relevant error information.

Reviewer: ```suggestion
This example shows how a user can extract the relevant error information.
```

          DummmyServer.serviceEndpointDummy()
        } catch {
          case e: StatusRuntimeException =>

            // Converting to a status object.
            val status = io.grpc.protobuf.StatusProto.fromThrowable(e)

            // Extracting error code id.
            assert(status.getCode == 10)

            // Extracting error message, both
            // machine oriented part: "MY_ERROR_CODE_ID(2,full-cor):",
            // and human oriented part: "A user oriented message".
            assert(status.getMessage == "MY_ERROR_CODE_ID(2,full-cor): A user oriented message")

            // Getting all the details
            val rawDetails: Seq[com.google.protobuf.Any] = status.getDetailsList.asScala.toSeq

            // Extracting error code id, error category id and optionally additional metadata.
            assert {
              rawDetails.collectFirst {
                case any if any.is(classOf[ErrorInfo]) =>
                  val v = any.unpack(classOf[ErrorInfo])
                  assert(v.getReason == "MY_ERROR_CODE_ID")
                  assert(v.getMetadataMap.asScala.toMap == Map("category" -> "2", "foo" -> "bar"))
              }.isDefined
            }

>             // Extracting full correlation id if present.

Reviewer: ```suggestion
            // Extracting full correlation id, if present.
```
